dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/ProcessInstanceServiceImpl.java

```diff
@@ -320,8 +320,11 @@ public Result queryProcessInstanceList(User loginUser, long projectCode, long pr
         }
 
         for (ProcessInstance processInstance : processInstances) {
-            processInstance.setDuration(
-                    DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime()));
+            // if processInstance is running, the endTime should be the current time
+            String duration = processInstance.getState() != null && processInstance.getState().isFinished() ?
+                    DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime()) :
+                    DateUtils.format2Duration(processInstance.getStartTime(), new Date());
```

Comment:
#in the method of DateUtils#format2Duration .when endTime is null will return new Date. i mean that maybe we can use 
```java
 DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime())
``` 
instead of
 ```java
String duration = processInstance.getState() != null && processInstance.getState().isFinished() ?
                    DateUtils.format2Duration(processInstance.getStartTime(), processInstance.getEndTime()) :
                    DateUtils.format2Duration(processInstance.getStartTime(), new Date()); 
```

Reply:
@fuchanghai no, if re-run, this endTime will not be null, so cannot use the processInstance.getEndTime()